streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKStreamImpl.java

```diff
@@ -237,10 +238,21 @@
                                                        + " grace=[" + windows.gracePeriodMs() + "],"
                                                        + " retention=[" + retentionPeriod + "]");
             }
-            supplier = Stores.persistentSessionStore(
-                materialized.storeName(),
-                Duration.ofMillis(retentionPeriod)
-            );
+
+            // get from default store implementation
+            StoreImplementation storeImplementation = materialized.storeImplementation();
+            if (storeImplementation != null) {
+                supplier = storeImplementation.sessionBytesStoreSupplier(
+                    materialized.storeName(),
+                    Duration.ofMillis(retentionPeriod));
+            } else {
+                // fall back to use Rocks Db by default if no store implementation provided
```

Comment:
Originally, when storeSupplier in materialized is null, we'll use `Rocks DB`. Now, we'll honor the `storeImplementation` setting (if we got props)